chapters/arrays.tex

```diff
@@ -463,9 +463,10 @@ \subsubsection{Reduction Expressions}\label{reduction-expressions}
 
 \subsection{Matrix and Vector Algebra Functions}\label{matrix-and-vector-algebra-functions}
 
-The following set of built-in matrix and vector algebra functions are available.  The function transpose and symmetric can be applied to any matrix.
-The functions \lstinline!outerProduct!, \lstinline!cross! and \lstinline!skew! require \lstinline!Real!/\lstinline!Integer! vector(s) or matrix as input(s)
-and returns a \lstinline!Real!/\lstinline!Integer! vector or matrix (the result is only \lstinline!Integer! if the input/all inputs are \lstinline!Integer!):
+The following set of built-in matrix and vector algebra functions are available.
+The function transpose can be applied to any matrix.
+The functions \lstinline!outerProduct!, \lstinline!symmetric!, \lstinline!cross! and \lstinline!skew! require \lstinline!Real!/\lstinline!Integer! vector(s) or matrix as input(s)
```

Comment:
Can't we simply rely on automatic conversion `Integer` to `Real` arrays, so that these well-behaved functions can have normal types both for inputs and outputs?

Reply:
That's how we implemented it in OpenModelica. I mean, it allows mixed Real/Integer vectors to cross, right? Should we update it?